src/Model/APContact.php

```diff
@@ -22,21 +23,30 @@ class APContact extends BaseObject
 	 * Resolves the profile url from the address by using webfinger
 	 *
 	 * @param string $addr profile address (user@domain.tld)
-	 * @return string url
+	 * @param string $url profile URL. When set then we return "true" when this profile url can be found at the address
+	 * @return string|boolean url
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
 	 */
-	private static function addrToUrl($addr)
+	private static function addrToUrl($addr, $url = null)
 	{
 		$addr_parts = explode('@', $addr);
 		if (count($addr_parts) != 2) {
 			return false;
 		}
 
+		$xrd_timeout = Config::get('system', 'xrd_timeout');
+
 		$webfinger = 'https://' . $addr_parts[1] . '/.well-known/webfinger?resource=acct:' . urlencode($addr);
 
-		$curlResult = Network::curl($webfinger, false, ['accept_content' => 'application/jrd+json,application/json']);
+		$curlResult = Network::curl($webfinger, false, ['timeout' => $xrd_timeout, 'accept_content' => 'application/jrd+json,application/json']);
 		if (!$curlResult->isSuccess() || empty($curlResult->getBody())) {
-			return false;
+			$webfinger = 'http://' . $addr_parts[1] . '/.well-known/webfinger?resource=acct:' . urlencode($addr);
```

Comment:
Please use `Strings::normaliseLink` instead.

Reply:
No. We have to check both. There are cases (I tested it with my data) where only the one or the other is working.

Reply:
I just meant about this line:
```suggestion
			$webfinger = Strings::normalizeLink($webfinger);
```